### DISCUSSION TITLE
Alembic tries to drop index when using TimescaleDB hypertable

### DISCUSSION THREAD

**Discussion by kimdre** [sqlalchemy/alembic#1465] [Usage Questions] [answered]
When I use TimescaleDB and create a hypertable on a normal PostgreSQL table, a new index on the given time column gets created automatically.
Since this index is not created from any migrations, Alembic tries to delete the index whenever I autogenerate a new migration.

Is there a way to tell Alembic to ignore this "external" index, e.g. by its name?

This is the migration content that always gets generated.

```python
def upgrade() -> None:
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_index('events_timestamp_idx', table_name='events')
    # ### end Alembic commands ###


def downgrade() -> None:
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_index('events_timestamp_idx', 'events', [sa.text('timestamp DESC')], unique=False)
    # ### end Alembic commands ###
```

**Comment by CaselIT** [comment]
Hi,

Sure, take a look at https://alembic.sqlalchemy.org/en/latest/api/runtime.html#alembic.runtime.environment.EnvironmentContext.configure.params.include_name

**Reply by kimdre** [reply, marked as answer]
Would this be the correct approach?

```python
def exclude_by_name(name, type_, parent_names):
    exclude_indexes = ["events_timestamp_idx"]
    if type_ == "index":
        return name not in exclude_indexes
    else:
        return True

context.configure(
    # ...
    include_name=exclude_by_name,
```

**Reply by CaselIT** [reply]
It should be yes

**Comment by MRigal** [comment]
By using the TimescaleDB dialect from this library, you may not have to ignore it anymore... https://github.com/dorosch/sqlalchemy-timescaledb

**Comment by AntiSol** [comment]
I believe I have solved this issue in my fork of the timescaledb dialect [here](https://github.com/AntiSol/sqlalchemy-timescaledb/commit/a109d8d94e0a4907fbe687c21df8013d564effcd). My approach is to intercept table creation and if the table is a hypertable, automatically add an index with the same settings added by timescaledb. As a result, alembic becomes aware of the existence of these indices, and so it doesn't try to drop them :)
